### DISCUSSION TITLE
Does uuid v1 have browser compatibility issues?

### DISCUSSION THREAD

**Discussion by PlayWithSanLei** [uuidjs/uuid#661] [Q&A] [answered]
I checked the information and learned that uuid version1 is generated based on timestamp and MAC address. Is there any browser compatibility problem? For example, some browsers cannot obtain the MAC address.

**Comment by broofa** [comment, marked as answer]
The `v1()` method works fine on all browsers. 

You're correct that browsers can't access the host's MAC address.  Instead, `uuid` uses a randomly generated value ([as allowed for by RFC 4122](https://www.ietf.org/rfc/rfc4122.html#section-4.5)).

This also avoids the potential security concerns associated with exposing host MAC addresses.

**Comment by PlayWithSanLei** [comment]
Thanks! That’s very helpful!

Robert Kieffer ***@***.***>于2022年11月25日 周五23:09写道：
